### PR TITLE
[v2] allow showMsgWhenDeprecated to accept an empty string as a parameter …

### PR DESCRIPTION
**Pull request by jace-roell**
**What It Does**
Modified showMsgWhenDeprecated() to allow an empty string as a paramter when no replacement is availible for deprecated command to print an alternative message.

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: packages/imperative/src/utilities/__tests__/CliUtils.unit.test.ts
Comment (ATorrise) [inline packages/imperative/src/utilities/__tests__/CliUtils.unit.test.ts:1]: @t1m0thyj does it make sense to make another test that checks that this alternate text isn't present when a non-empty string is passed to showMsgWhenDeprecated? probably unnecessary

**Review by ATorrise — APPROVED** [review]

File: packages/imperative/src/utilities/__tests__/CliUtils.unit.test.ts [hunk 1]
```diff
@@ -320,6 +320,13 @@ describe("CliUtils", () => {
             CliUtils.showMsgWhenDeprecated(handlerParms);
             expect(responseErrText).toEqual(notSetYet);
         });
+
+        it("should produce alternative text when deprecatedReplacement is an empty string", () => {
```
Comment (ATorrise) [inline packages/imperative/src/utilities/__tests__/CliUtils.unit.test.ts:324]: @t1m0thyj does it make sense to make another test that checks that this alternate text isn't present when a non-empty string is passed to showMsgWhenDeprecated? probably unnecessary

Comment (t1m0thyj) [reply packages/imperative/src/utilities/__tests__/CliUtils.unit.test.ts:324]: We could add an `expect` statement to one of the tests above to verify that the output does not contain the alternate text 😋

**Review by awharn — APPROVED** [review]

File: packages/imperative/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 All notable changes to the Imperative package will be documented in this file.
 
+
+## Recent Changes
+
+- BugFix: Modified showMsgWhenDeprecated() to allow an empty string as a paramter when no replacement is availible for deprecated command to print an alternative message. [#2041](https://github.com/zowe/zowe-cli/issues/2041)
```
Comment (anaxceron) [inline packages/imperative/CHANGELOG.md:8]: - Please say what `showMsgWhenDeprecated()` is: Modified the `showMsgWhenDeprecated()` **command/property/cat breed** ...
- Fix typo in **paramter**
- This part of the sentence sounds a little awkward, not sure what you're saying: "for deprecated command to print an alternative message." Do you mean "for **_the_** deprecated command"?

**Review by anaxceron — COMMENTED** [review]
Requested a few edits - thank you!

**Review by anaxceron — APPROVED** [review]
Looks good -- thanks @jace-roell!

File: packages/imperative/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,11 @@
 
 All notable changes to the Imperative package will be documented in this file.
 
+
+## Recent Changes
+
+- BugFix: Modified showMsgWhenDeprecated() function to allow an empty string as a parameter when no replacement is availible for the deprecated command. When no replacement is availible an alternative message will be printed. [#2041](https://github.com/zowe/zowe-cli/issues/2041)
```
Comment (t1m0thyj) [inline packages/imperative/CHANGELOG.md:8]: Please fix typo and use code block when referencing class or method names 🙂 
```suggestion
- BugFix: Modified `showMsgWhenDeprecated` function to allow an empty string as a parameter when no replacement is available for the deprecated command. When no replacement is available an alternative message will be printed. [#2041](https://github.com/zowe/zowe-cli/issues/2041)
```

File: packages/imperative/src/utilities/src/CliUtils.ts
Comment (t1m0thyj) [inline packages/imperative/src/utilities/src/CliUtils.ts:1]: Changes in this file look good, but doesn't make the command appear as deprecated in CLI help.

Should we update the help generator logic as well?
https://github.com/zowe/zowe-cli/blob/c5c5df17edee1cf44e351a58925d407607c93f02/packages/imperative/src/cmd/src/help/DefaultHelpGenerator.ts#L225-L227

If we change the `if` check to `if (command.deprecatedReplacement != null)`, this would make "(deprecated)" appear for commands with an empty `deprecatedReplacement` string.

**Review by awharn — APPROVED** [review]

File: packages/imperative/src/utilities/src/CliUtils.ts [hunk 4]
```diff
@@ -378,20 +378,26 @@ export class CliUtils {
...
+        let oldCmd: string;
+        if (handlerParms.definition.deprecatedReplacement || handlerParms.definition.deprecatedReplacement === "") {
             // form the command that is deprecated
-            let oldCmd: string;
             if (handlerParms.positionals.length >= 1) {
                 oldCmd = handlerParms.positionals[0];
             }
             if (handlerParms.positionals.length >= 2) {
                 oldCmd = oldCmd + " " + handlerParms.positionals[1];
             }
```
Comment (t1m0thyj) [inline packages/imperative/src/utilities/src/CliUtils.ts:389]: If a single command (rather than a command group) is deprecated, the message may not show the full command. For example, I deprecated the "zowe zcsp issue cwp" command in the CLI sample plug-in and it shows the following message:
```
Warning: The command 'zcsp issue' is deprecated.
Obsolete component. No replacement exists
```

If we show all positionals, then the full command name will be displayed (e.g. `zowe zcsp issue cwp`):
```suggestion
            const oldCmd = handlerParms.positionals.join(" ");
```

Comment (jace-roell) [reply packages/imperative/src/utilities/src/CliUtils.ts:389]: I think that bug may have been there all along, good to know though. Thanks @t1m0thyj

File: packages/imperative/src/utilities/src/CliUtils.ts [hunk 5]
```diff
@@ -378,20 +378,21 @@ export class CliUtils {
      * @memberof CliUtils
      */
     public static showMsgWhenDeprecated(handlerParms: IHandlerParameters) {
-        if (handlerParms.definition.deprecatedReplacement) {
```
Comment (awharn) [inline packages/imperative/src/utilities/src/CliUtils.ts:381]: This line should be removed, since oldCmd is declared as a const on L384.

**Review by awharn — CHANGES_REQUESTED** [review]

**Review by awharn — APPROVED** [review]
Thanks @jace-roell! Sorry for all the change requests. 🚀

**Review by gejohnston — APPROVED** [review]
